### ISSUE TITLE
Add support for Cuelang

### ISSUE THREAD

**Issue by leoluk** [github/linguist#5013]
## Preliminary Steps

Please confirm you have...
- [X] reviewed [How Linguist Works](https://github.com/github/linguist#how-linguist-works),
- [X] reviewed the [Troubleshooting](https://github.com/github/linguist#troubleshooting) docs,
- [X] considered implementing an [override](https://github.com/github/linguist#overrides),
- [X] verified an issue has not already been logged for your issue ([linguist issues](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+repo%3Agithub/linguist)).

## Problem Description

Linguist should support for https://cuelang.org/docs/about, an upcoming data validation/configuration language that is already seeing widespread use.

Cue files are currently hard to read on GitHub: https://github.com/leoluk/NetMeta/blob/master/deploy/single-node/traefik/deployment.cue

There's a WIP pull request by @pchaigno which seems blocked by an unfortunate file extension collision that'll require heuristics: https://github.com/github/linguist/pull/4941

**Comment by myitcv** [comment]
@lildude is there anything we can do to help move this one along?

**Comment by lildude** [comment]
@myitcv As https://github.com/github/linguist/pull/4941 appears to have been abandoned, the best thing to do is open a new PR, but this will need to include support for Cuelang and the other language identified. 

From a quick look, it looks like all the other `.cue` files may be [cue sheet](https://en.wikipedia.org/wiki/Cue_sheet_(computing)) which is quite different so adding both in the same PR shouldn't be too challenging. A grammar isn't needed for inclusion.

**Comment by myitcv** [comment]
Thanks @lildude. 

@zchee - you mentioned offline you'd be keen to work on this - did you make any progress? I'm happy to help out.

**Comment by zchee** [comment]
@myitcv ah, I forgot it. will send PR.

**Comment by lildude** [comment]
https://github.com/github/linguist/pull/5312 has been merged.

It just missed the last release so GitHub.com won't gain support until the next release is made which will be in about a month's time.  As Linguist now has support, I'm going to close this issue. Thanks to everyone involved 🙇.
